### PR TITLE
Revert "optimize softmax with cross entropy hard label"

### PR DESCRIPTION
**Pull request by Xreki**
### PR types
Performance optimization

### PR changes
Ops

### Describe
Reverts PaddlePaddle/Paddle#32290

#32290 引起了ResNet50训练精度的下降，训练13个epoch，相比2.1版本，ResNet50训练精度从0.52838下降到了0.50610。

固定所有的随机量使用小数据集跑5个epoch，监控loss；这个pr合入前的loss是6.80262；合入后loss是 6.77933，目前最新的develop loss也是 6.77933；

测试方法：

```
export FLAGS_cudnn_deterministic=True
sed -i 's/RandCropImage/ResizeImage/g' configs/ResNet/ResNet50.yaml
sed -ie '/RandFlipImage/d' configs/ResNet/ResNet50.yaml
sed -ie '/flip_code/d' configs/ResNet/ResNet50.yaml
python -m paddle.distributed.launch tools/train.py -c configs/ResNet/ResNet50.yaml
```

### CODE REVIEW

**Review by zhangting2020 — APPROVED** [review]

**Review by chenwhql — APPROVED** [review]
LGTM for PADDLE_ENFORCE
